### ISSUE TITLE
False positives with php-http/* packages

### ISSUE THREAD

**Issue by Jean85** [composer-unused/composer-unused#337]
## Describe the bug
I have a small project where I leverage HTTPlug packages to stay agnostic about HTTP clients. I recently bumped the minimum requirement from `^7.2` to `^7.4`, and bumped `composer-unused` from `^0.7` to the latest version, `^0.8.1`.

It now reports false positives against all `php-http/*` packages:
```
 Unused packages
 ✗ php-http/client-common
 ✗ php-http/discovery
 ✗ php-http/httplug
 ✗ php-http/message-factory
```
Despite having this piece of code that would (at the very least) remove `php-http/discovery` from that list:
```php
use Http\Discovery\HttpClientDiscovery;
use Http\Discovery\Psr17FactoryDiscovery;
// ...

    public function create(array $options): Client
    {
        $this->requestFactory = $this->requestFactory ?? Psr17FactoryDiscovery::findRequestFactory();
        $this->streamFactory = $this->streamFactory ?? Psr17FactoryDiscovery::findStreamFactory();
        $this->uriFactory = $this->uriFactory ?? Psr17FactoryDiscovery::findUrlFactory();
        $this->httpClient = $this->httpClient ?? HttpClientDiscovery::find();
        // ...
```

**Comment by Jean85** [comment]
I'm trying to debug it further, and it seems that `\ComposerUnused\ComposerUnused\Composer\Package` doesn't contain any autoload information, despite reading from packages with correct PSR-4 configurations: https://github.com/php-http/discovery/blob/98ead1fe1c3af88bd65d927b8bbc9fdf1ae83423/composer.json#L26-L29

FTR, I also updated to Composer 2.2.7. I'll debug further.

**Comment by Jean85** [comment]
Ok this seems already fixed in #318. Closing as duplicate of #316.

**Comment by icanhazstring** [comment]
Nice, thanks for still digging into it :)
Hopefully helpful for others who stumble upon this.

Also release coming soon which fixes this as well.
